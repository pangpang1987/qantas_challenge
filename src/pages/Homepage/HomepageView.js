import React from "react";
import PropTypes from "prop-types";



import {HomepageContainer, HomepageBox, Title, SubTitle, Link} from './styled';

class HomepageView extends React.Component {
  componentDidMount() {
    this.props.actions.fetchHomepage();
  }

  render() {
    const { isLoading, homepage } = this.props;

    if(isLoading) {
      return <div>Loading...</div>
    }
    const {heroImageUrl, heading, subheading} = homepage;

    return <HomepageContainer image={heroImageUrl}>
      <HomepageBox>
        <Title>{heading}</Title>
        <SubTitle>{subheading}</SubTitle>
        <Link to="/faqs">Learn more</Link>
      </HomepageBox>

    </HomepageContainer>;
  }
}

HomepageView.propTypes = {
  isLoading: PropTypes.bool,
  homepage: PropTypes.object,
  actions: PropTypes.object
};

export default HomepageView;
