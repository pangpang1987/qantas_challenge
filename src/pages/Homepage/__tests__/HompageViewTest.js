import React from "react";

import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import HomepageView from "../HomepageView";

describe("Component HomepageView", () => {
  const fetchHomepage = jest.fn();
  const actions = { fetchHomepage };
  const homepage = {
    heading: "Heading",
    subheading: "Sub heading",
    heroImageUrl: "https://developers.google.com/identity/images/incorrect-buttons/lightbutton.png"
  };

  it("should render initial state properly", () => {
    const component = shallow(<HomepageView homepage={{}} isLoading={false} actions={actions} />);

    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
    expect(fetchHomepage.mock.calls.length).toBe(1);
  });

  it("should render a loader when fetching data", () => {
    const component = shallow(<HomepageView homepage={{}}isLoading={true} actions={actions} />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });

  it("should render properly with data", () => {
    const component = shallow(<HomepageView homepage={homepage} isLoading={false} actions={actions} />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });

});
