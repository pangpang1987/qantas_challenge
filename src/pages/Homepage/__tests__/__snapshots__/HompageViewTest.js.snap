// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component HomepageView should render a loader when fetching data 1`] = `
<div>
  Loading...
</div>
`;

exports[`Component HomepageView should render initial state properly 1`] = `
<styled.div>
  <styled.div>
    <styled.h1 />
    <styled.p />
    <Styled(Link)
      to="/faqs"
    >
      Learn more
    </Styled(Link)>
  </styled.div>
</styled.div>
`;

exports[`Component HomepageView should render properly with data 1`] = `
<styled.div
  image="https://developers.google.com/identity/images/incorrect-buttons/lightbutton.png"
>
  <styled.div>
    <styled.h1>
      Heading
    </styled.h1>
    <styled.p>
      Sub heading
    </styled.p>
    <Styled(Link)
      to="/faqs"
    >
      Learn more
    </Styled(Link)>
  </styled.div>
</styled.div>
`;
